conans/client/generators/cmake_find_package_multi.py

```diff
@@ -184,8 +184,8 @@ class CMakeFindPackageMultiGenerator(CMakeFindPackageGenerator):
             include(${f})
         endforeach()
 
-        if({{ pkg_name }}_FIND_COMPONENTS)
-            foreach(_FIND_COMPONENT {{ '${'+pkg_name+'_FIND_COMPONENTS}' }})
+        if({{ pkg_filename }}_FIND_COMPONENTS)
+            foreach(_FIND_COMPONENT {{ '${'+pkg_filename+'_FIND_COMPONENTS}' }})
```

Comment:
Same situation as https://github.com/conan-io/conan/pull/10077/files#r756964886 this was not working if cpp_info.filename was different from names